src/main/java/org/sunbird/assessment/service/AssessmentServiceV2Impl.java

```diff
@@ -35,195 +40,364 @@ public class AssessmentServiceV2Impl implements AssessmentServiceV2 {
+                                        if (!ObjectUtils.isEmpty(questionMap)) {
+                                            Boolean isInsertedToRedis = redisCacheMgr.putCache(Constants.QUESTION_ID + question.get(Constants.IDENTIFIER), question);
+                                            questionList.add(filterQuestionMapDetail(question));
+                                        } else {
+                                            errMsg = "Failed to get Question Details for Id: %s";
+                                            logger.error(String.format("Failed to get Question Details for Id: %s", question.get(Constants.IDENTIFIER).toString()));
+                                            break;
+                                        }
+                                    }
+                                } else {
+                                    logger.error(String.format("Failed to get Question Details from the Question List API for the IDs: %s", newIdentifierList.toString()));
+                                }
+                            }
+                            if (!questionList.isEmpty() && errMsg.isEmpty()) {
+                                response.getResult().put(Constants.QUESTIONS, questionList);
+                            }
+                            else
+                            {
+                                errMsg = "The list of questions are Empty/Invalid";
+                            }
                         } else {
-                            logger.error(new Exception("Failed to get Question Details for Id: " + qmap.get(Constants.IDENTIFIER)));
+                            errMsg = "The Questions Ids Provided are not a part of the active user assessment session";
                         }
+                    } else {
+                        errMsg = "Please provide a valid assessment Id/Session Expired";
                     }
                 }
+            } else {
+                errMsg = "User Id doesn't exist! Please supply a valid auth token";
             }
-            return prepareQuestionResponse(questionList, questionList.size() > 0);
         } catch (Exception e) {
-            logger.error(e);
-            throw new ApplicationLogicError("REQUEST_COULD_NOT_BE_PROCESSED", e);
+            logger.error(String.format("Exception in %s : %s", "get Question List", e.getMessage()), e);
+            errMsg = "Failed to fetch the question list. Exception: " + e.getMessage();
+        }
+        if (StringUtils.isNotBlank(errMsg)) {
+            response.getParams().setStatus(Constants.FAILED);
+            response.getParams().setErrmsg(errMsg);
+            response.setResponseCode(HttpStatus.BAD_REQUEST);
         }
+        return response;
+
+    }
 
+    private static String validateAuthTokenAndFetchUserId(String authUserToken) {
+        return RequestInterceptor.fetchUserIdFromAccessToken(authUserToken);
     }
```

Comment:
Separate method is not required for single lines -- previous comment w.r.t. validation method is for validating incoming request object.
you can move this line also to the other validation method... check the userId is not empty in that method and return errMsg...